src/Uno.Analyzers/UnoNotImplementedAnalyzer.cs

```diff
@@ -78,9 +82,63 @@ private void OnObjectCreationExpression(SyntaxNodeAnalysisContext contextAnalysi
 			}
 		}
 
-		private static bool HasNotImplementedAttribute(INamedTypeSymbol notImplementedSymbol, ISymbol namedSymbol)
+		private string[] GetDirectives(SyntaxNodeAnalysisContext contextAnalysis)
 		{
-			return namedSymbol.GetAttributes().Any(a => a.AttributeClass == notImplementedSymbol);
+			var directives = contextAnalysis.Node.GetLocation()?.SourceTree.Options.PreprocessorSymbolNames.ToArray() ?? new string[0];
+
+			if (directives.Length == 0)
+			{
+				var directive = contextAnalysis
+					.Node
+					.GetLocation()
+					.SourceTree
+					.GetRoot()
+					.GetFirstDirective() as DefineDirectiveTriviaSyntax;
```

Comment:
I'm curious what this will do?

Reply:
I'll document. It's only used during tests, as in normal compilations, there are (almost) always symbols.